### PR TITLE
mail/rspamd: add whitelisting to greylist

### PR DESCRIPTION
**Pull request by mimugmail**

### CODE REVIEW

File: mail/rspamd/pkg-descr [hunk 1]
```diff
@@ -1,3 +1,10 @@
 Rspamd is fast, modular and lightweight spam filter. It is designed to work
 with big amount of mail and can be easily extended with own filters written in
 lua.
+
+Plugin Changelog
+----------------
+
+1.5
+
+* Add whitelisting to Greylog section
```
Comment (fabianfrz) [inline mail/rspamd/pkg-descr]:
```suggestion
* Add whitelisting to Graylist section
```

I always write in AE, not BE, also fix the typo.

File: mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml [hunk 2]
```diff
@@ -138,6 +138,14 @@
                 <type>text</type>
                 <help>Mask bits are used to limit the network range from which the message may be resent. This is used to avoid a rejection if another server sends the second mail.</help>
             </field>
+            <field>
+                <id>rspamd.graylist.whitelist_ip</id>
+                <label>Whitelisted IPs</label>
+                <type>select_multiple</type>
+                <style>tokenize</style>
+                <allownew>true</allownew>
+                <help>Mails from IPs or networks in CIDR format entered here will be whitelisted from greylisting.</help>
```
Comment (fabianfrz) [inline mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml]:
```suggestion
                <help>Mails from IPs or networks in CIDR format entered here will be whitelisted from graylisting.</help>
```

see top (also match the id)

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml [hunk 3]
```diff
@@ -173,6 +173,9 @@
         <default>64</default>
         <ValidationMessage>A valid IPv6 mask must be between 1 and 128 bits. 64 bits are recommended as this is the recommended subnet size in IPv6.</ValidationMessage>
       </ipv6mask>
+      <whitelist_ip type="CSVListField">
+        <Required>N</Required>
```
Comment (fabianfrz) [inline mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: maybe add a regex here to match a-f, 0-9, "." and ":" (plus the "," for the CSV).

File: mail/rspamd/src/opnsense/service/templates/OPNsense/Rspamd/+TARGETS [hunk 4]
```diff
@@ -10,6 +10,7 @@ spamtrap.conf:/usr/local/etc/rspamd/local.d/spamtrap.conf
 surbl-whitelist.inc.local:/var/db/rspamd/surbl-whitelist.inc.local
 2tld.inc.local:/var/db/rspamd/2tld.inc.local
 greylist.conf:/usr/local/etc/rspamd/local.d/greylist.conf
+greylist_ip.wl:/usr/local/etc/rspamd/local.d/greylist_ip.wl
```
Comment (fabianfrz) [inline mail/rspamd/src/opnsense/service/templates/OPNsense/Rspamd/+TARGETS]:
```suggestion
graylist_ip.wl:/usr/local/etc/rspamd/local.d/graylist_ip.wl
```

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml (hunk 3 above)
Comment (mimugmail) [reply mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: Good one, done

**Comment by fabianfrz** [comment]
@mimugmail merged - thanks for the contribution.

**Pull request merged**